layout/ComponentFallbackQuery.js

```diff
@@ -0,0 +1,527 @@
+    o.bool.must.push(addSecondary(
+      vs.var('input:locality').toString(),
+      [
+        'parent.locality',
+        'parent.locality_a',
+        'parent.localadmin',
+        'parent.localadmin_a'
+      ]
+    ));
+  }
+}
+
+function addSecCounty(vs, o) {
+  // add county if specified
+  if (vs.isset('input:county')) {
+    o.bool.must.push(addSecondary(
+      vs.var('input:county').toString(),
+      [
+        'parent.county',
+        'parent.county_a',
+        'parent.macrocounty',
+        'parent.macrocounty_a'
+      ]
+    ));
+  }
+}
+
+function addSecRegion(vs, o) {
+  // add region if specified
+  if (vs.isset('input:region')) {
+    o.bool.must.push(addSecondary(
+      vs.var('input:region').toString(),
+      [
+        'parent.region',
+        'parent.region_a',
+        'parent.macroregion',
+        'parent.macroregion_a'
+      ]
+    ));
+  }
+}
+
+function addSecCountry(vs, o) {
+  // add country if specified
+  if (vs.isset('input:country')) {
+    o.bool.must.push(addSecondary(
+      vs.var('input:country').toString(),
+      [
+        'parent.country',
+        'parent.country_a',
+        'parent.dependency',
+        'parent.dependency_a'
+      ]
+    ));
+  }
+}
+
+
+function addQuery(vs) {
+  var o = addPrimary(
+    vs.var('input:query').toString(),
+    'venue',
+    [
+      'phrase.default'
+    ],
+    false
+  );
+
+  addSecNeighbourhood(vs, o);
+  addSecBorough(vs, o);
+  addSecLocality(vs, o);
+  addSecCounty(vs, o);
+  addSecRegion(vs, o);
+  addSecCountry(vs, o);
+
+  return o;
+
+}
+
+function addAddress(vs) {
+  var o = {
+    bool: {
+      _name: 'fallback.address',
+      must: [],
+      should: [],
+      filter: {
+        bool: {
+          must: [
+            {
+              term: {
+                layer: 'address'
+              }
+            },
+            {
+              query_string: {
+                default_field: 'name.default',
+                default_operator: 'AND',
+                analyzer: 'peliasQueryFullToken',
+                query: vs.var('input:address').toString()
+              }
+            }
+          ]
+
+        }
+      }
+    }
+  };
+
+  // THIS VALUE IS OPEN TO DEBATE AND CONFIGURATION
+  o.bool.boost = 100;
+
+  // if (vs.isset('boost:street')) {
+  //   o.bool.boost = vs.var('boost:street');
+  // }
+
+  addSecPostCode(vs, o);
+  addSecNeighbourhood(vs, o);
+  addSecBorough(vs, o);
+  addSecLocality(vs, o);
+  addSecCounty(vs, o);
+  addSecRegion(vs, o);
+  addSecCountry(vs, o);
+
+  return o;
+
+}
+
+function addStreet(vs) {
+  var o = {
+    bool: {
+      _name: 'fallback.street',
+      must: [
+        {
+          match_phrase: {
+            'address_parts.street': vs.var('input:address').toString()
+          }
+        }
+      ],
+      should: [],
+      filter: {
+        term: {
+          layer: 'street'
+        }
+      }
+    }
+  };
+
+  // THIS VALUE IS OPEN TO DEBATE AND CONFIGURATION
+  o.bool.boost = 10000;
```

Comment:
this value feels really high. i remember the elasticsearch book suggesting boosts never go above some relatively small number (I think it was either 10 or 100), and that if higher values are needed its probably best to fix things another way.